### PR TITLE
Support exodus payload parsing for sigma spend transaction

### PR DESCRIPTION
**Pull request by thebevrishot**
https://trello.com/c/O7Ab5HzU/528-update-transaction-parser-for-exodus-to-be-able-to-parse-sigma-spend-transaction-55

### CODE REVIEW

File: src/exodus/test/exodus_handler_tx.cpp [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+BOOST_FIXTURE_TEST_SUITE(exodus_handler_tx_tests, ZerocoinTestingSetup200)
+
+static std::vector<unsigned char> createMockSpendPayload()
+{
+    std::vector<unsigned char> payload;
+
+    uint16_t messageType = 1024;
+    uint16_t messageVer = 0;
+    exodus::swapByteOrder16(messageVer);
+    exodus::swapByteOrder16(messageType);
```
Comment (ultimaweapon) [inline src/exodus/test/exodus_handler_tx.cpp]: Actually we don't need to swap bytes due to we support only little endian.

File: src/exodus/test/exodus_handler_tx.cpp [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+    PUSH_BACK_BYTES(payload, messageVer);
+    PUSH_BACK_BYTES(payload, messageType);
+
+    return payload;
+}
+
+CBlock getBlockHeighest() {
```
Comment (ultimaweapon) [inline src/exodus/test/exodus_handler_tx.cpp]: The brace should consistent. So it should on a next line for a function.

File: src/exodus/exodus.cpp [hunk 3]
```diff
@@ -815,9 +821,20 @@ static int parseTransaction(bool bRPConly, const CTransaction& wtx, int nBlock,
     LOCK(cs_tx_cache);
 
     // Add previous transaction inputs to the cache
-    if (!FillTxInputCache(wtx)) {
-        PrintToLog("%s() ERROR: failed to get inputs for %s\n", __func__, wtx.GetHash().GetHex());
-        return -101;
+    switch (inputMode) {
```
[the remark this answers is not in the document]
Comment (thebevrishot) [reply src/exodus/exodus.cpp]: make senses.

File: src/exodus/exodus.cpp [hunk 4]
```diff
@@ -865,7 +882,13 @@ static int parseTransaction(bool bRPConly, const CTransaction& wtx, int nBlock,
         // NEW LOGIC - the sender is chosen based on the first vin
 
         // determine the sender, but invalidate transaction, if the input is not accepted
-        {
+
+        // do not need to check sigma.
+        switch (inputMode) {
```
[the remark this answers is not in the document]
Comment (thebevrishot) [reply src/exodus/exodus.cpp]: make sense.

File: src/exodus/test/exodus_handler_tx.cpp (hunk 1 above)
Comment (thebevrishot) [reply src/exodus/test/exodus_handler_tx.cpp]: nice catch

Comment (thebevrishot) [reply src/exodus/test/exodus_handler_tx.cpp]: after personal discussion, keep them.

File: src/exodus/tx.h [hunk 5]
```diff
@@ -44,6 +44,8 @@ class CMPTransaction
     unsigned int type;
     unsigned short version; // = MP_TX_PKT_V0;
 
+    exodus::InputMode inputMode;
```
[the remark this answers is not in the document]
Comment (thebevrishot) [reply src/exodus/tx.h]: fixed

**Pull request merged**